Blog: Quantum Computing April Fools (#2949)
FusionAuth/fusionauth-site: astro/src/content/blog/quantum-computing.mdx

---
publish_date: 2024-04-01
title: "FusionAuth Cracks the Code: Quantum Passwords and a Universe of Secure Logins"
description: How FusionAuth has built the world's first Quantum Vault
authors: Alex Patterson

> image: /img/blogs/2023-hacktoberfest/hacktoberfest-2023.png

Reviewer: Needs a header (but you probably knew that :) )

categories: News
tags: community story, april fools
excerpt_separator: "{/* more */}"
---

> Ever since the show [Devs](https://www.wired.com/story/inside-devs-dreamy-silicon-valley-quantum-thriller/) came out on FX in 2020, FusionAuth’s CEO Brian Pontarelli started thinking about the possibilities of adding quantum computing as part of FusionAuth’s product offering. We added a Skunkworks-style team that went into deep secrecy surrounding the project. After four years of iterating on both the physical design and the software to accompany the solution we're thrilled to announce a revolutionary new advancement in the realm of user authentication: **The FusionAuth Quantum Vault**. 

Reviewer: ```suggestion
Ever since the show [Devs](https://www.wired.com/story/inside-devs-dreamy-silicon-valley-quantum-thriller/) came out on FX in 2020, FusionAuth’s CEO Brian Pontarelli has been musing on the possibilities of adding quantum computing to FusionAuth. Four years ago, FusionAuth commissioned a Skunkworks that went into deep secrecy and worked on this concept. After years of iterating on both the physical design and the software to accompany the solution we're thrilled to announce a revolutionary new advancement in the realm of user authentication: **The FusionAuth Quantum Vault**. 
```

{/* more */}

## How Quantum Vulnerabilities Impact JWTs

> [JSON Web Tokens](/articles/tokens/) (JWTs) are a popular way to securely share information online.  However, traditional JWTs that rely on certain encryption methods could be broken by powerful quantum computers. These attacks primarily apply to JWTs that use asymmetric cryptography – where your login info is validated against a public key. The issue?  Future quantum computers could potentially break these encryption methods.

Reviewer: ```suggestion
[JSON Web Tokens](/articles/tokens/) (JWTs) are a popular way to securely share information online.  However, traditional JWTs that rely on certain encryption methods could be broken by powerful quantum computers. These attacks primarily apply to JWTs that use asymmetric cryptography – where your login info is validated against a public key. The issue? Future quantum computers can theoretically break these encryption methods by using alternate pathways to, for example, factor large prime numbers.
```

## The FusionAuth Quantum Vault Solution

The FusionAuth Quantum Vault reimagines how JWTs could work in the quantum era. 

Here's what it offers:

> * **Advanced Quantum-Resistant Algorithms:** Instead of relying on traditional encryption, the FusionAuth Quantum Vault will explore next-generation, quantum-resistant algorithms.  This ensures that even if quantum computers become widespread, your JWTs will remain secure.

Reviewer: ```suggestion
* **Advanced Quantum-Resistant Algorithms:** Instead of relying on traditional encryption, the FusionAuth Quantum Vault uses next-generation, quantum-resistant algorithms.  This ensures that even if quantum computers become widespread, your JWTs will remain secure.
```

> * **Beyond Just Encryption:** While quantum-safe encryption is vital, the FusionAuth Quantum Vault goes further. We believe the future of authentication lies in minimizing the amount of sensitive data we need to store anywhere. We'll investigate more stateful token approaches that reduce what needs to be protected in the first place.

Reviewer: ```suggestion
* **Beyond Just Encryption:** While quantum-safe encryption is vital, the FusionAuth Quantum Vault goes further. We believe the future of authentication lies in minimizing the amount of sensitive data customers need to store anywhere. Quantum entanglement reduces the need for signing keys.
```

> 

Reviewer: ```suggestion
* **Password superposition** When used carefully, the concept of superposition can offer efficiencies, allowing users to log in to multiple applications at one time. Or maybe letting many users log in to one application at once. We're not quite sure, we didn't really understand the double slit experiment.

```

> The FusionAuth Quantum Vault doesn't just patch a hole; it aims to rethink how JWTs can evolve for a more secure future. We're looking beyond just encryption and exploring new token models to keep your data safe in the face of evolving threats. 

Reviewer: ```suggestion
The FusionAuth Quantum Vault doesn't just patch a hole; it aims to rethink how JWTs and authentication can evolve for a more secure future. We're looking beyond just encryption and exploring new token models to keep your data safe in the face of evolving threats. 
```

> ## What in the qubit?

Reviewer: ```suggestion
## What In The Qubit?
```

Should be capitalized.

> You might be thinking, "Quantum computers? Those things are still in their infancy, barely clinging to the shores of theoretical physics!" But fear not, intrepid reader, because the brilliant minds at FusionAuth have been working tirelessly in our secret lab (complete with beanbag chairs and a kombucha fountain, of course) to develop a quantum authentication system that would make even Schrödinger's cat do a double meow.

Reviewer: ```suggestion
You might be thinking, "Quantum computers? Those things are still in their infancy, barely clinging to the shores of theoretical physics!" But fear not, intrepid reader, because the brilliant minds at FusionAuth have been working tirelessly in our secret lab (complete with beanbag chairs and a kombucha fountain, of course) to develop a quantum authentication system that would make even Schrödinger's cat do a double meow. (Yeah, the cat lives.)
```

> ## So, how does this quantum contraption work?

Author: ```suggestion
## So, How Does This Quantum Contraption Work?
```

> ## So, how does this quantum contraption work?

Reviewer: Capitalize.

Imagine a password, not as a string of characters, but as a probability wave function. In the quantum realm, a password can exist in multiple states simultaneously, until the act of logging in collapses the wave function into a single, verified state. This means that brute-force attacks become a thing of the past, as hackers would be grappling with not just one password, but a superposition of infinite possibilities.

## But wait, there's more!

> The FusionAuth Quantum Vault isn't just about impenetrable security. It's also about scalability. With the mind-bending power of qubits, our system can handle an unimaginable number of users – **up to 40 billion, to be precise**. That's more than the entire population of the Earth! So, whether you're a social media platform with millions of users or a niche cat video forum with a dedicated following of twelve, the FusionAuth Quantum Vault can handle your login needs.

Reviewer: ```suggestion
The FusionAuth Quantum Vault isn't just about impenetrable security. It's also about scalability. With the mind-bending power of qubits, our system can handle an unimaginable number of users – **up to 40,024,291,137, to be precise**. That's more than the entire population of the Earth! So, whether you're a social media platform with millions of users or a niche cat video forum with a dedicated following of twelve, the FusionAuth Quantum Vault can handle your login needs.
```

## The Ethical Implications of Quantum Authentication

> Of course, with great power comes great responsibility. We understand that the idea of a quantum computer wielding the keys to billions of passwords might raise some eyebrows. But fear not, privacy advocates! The FusionAuth Quantum Vault operates on a zero-knowledge principle. This means that your password never leaves your device in its original form. Instead, it's transformed into a cryptographic hash that only you and the vault can understand.

Reviewer: ```suggestion
Of course, with great power comes great responsibility. We understand that the idea of a quantum computer wielding the keys to billions of passwords might raise some eyebrows. But fear not, privacy advocates! The FusionAuth Quantum Vault operates on a zero-knowledge principle. This means that your password never leaves your device in its original form. Instead, it's transformed into a cryptographic hash that only you and the vault can use.
```

> Google has most recently published an [article](https://bughunters.google.com/blog/5108747984306176/google-s-threat-model-for-post-quantum-cryptography) outlining the fact, "if we do not encrypt our data with a quantum-secure algorithm right now, an attacker who is able to store current communication will be able to decrypt it in as soon as a decade". This was exactly our motivation for making one of the most secure authentication systems to utilize **The FusionAuth Quantum Vault** so that your data will remain secure for the foreseeable future. 

Reviewer: ```suggestion
Google has most recently published an [article](https://bughunters.google.com/blog/5108747984306176/google-s-threat-model-for-post-quantum-cryptography) outlining the fact, "if we do not encrypt our data with a quantum-secure algorithm right now, an attacker who is able to store current communication will be able to decrypt it in as soon as a decade". This was exactly our motivation for making one of the most secure authentication systems to utilize **The FusionAuth Quantum Vault** so that your data will remain secure for the foreseeable future.

Good one Google! You're only four years behind, keep trying!
```

## When Will The FusionAuth Quantum Vault be Available?

**The FusionAuth Quantum Vault** will change the world of online security as we know it. We are currently running a googolplexianth number of tests on our prototype vault pictured below.

![Photo of FusionAuth Quantum Vault](/img/blogs/quantum-computing/Gemini_Generated_Image.jpeg)

> For now, consider this a glimpse into the future of authentication – a future where logins are instantaneous, secure, and scalable beyond imagination. 

Reviewer: ```suggestion
For now, consider this a glimpse into the future of authentication – a future where JWTs are safe and logins are instantaneous, secure, and scalable beyond imagination. 
```

**P.S.**

> While the FusionAuth Quantum Vault might not be here just yet, we are constantly innovating to bring you the most cutting-edge authentication solutions available. Check out our latest [resources](/resource/all) to learn more about how FusionAuth can keep your data safe and secure.

Reviewer: ```suggestion
While the FusionAuth Quantum Vault might not be here just yet, we are constantly innovating to bring you the most cutting-edge authentication solutions available. Check out our latest [resources](/resource/all) to learn more about how FusionAuth can keep your data safe and secure.

**P.P.S.** Happy April Fools!
```
